config/src/network_id.rs

```diff
@@ -1,12 +1,62 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
+use crate::config::RoleType;
+use libra_types::PeerId;
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 
+/// A grouping of common information between all networking code for logging.
+/// This should greatly reduce the groupings between these given everywhere, and will allow
+/// for logging accordingly.  TODO: Figure out how to split these as structured logging
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct NetworkContext {
+    network_id: NetworkId,
+    role: RoleType,
+    peer_id: PeerId,
+}
+
+impl std::fmt::Display for NetworkContext {
+    fn fmt(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(
+            formatter,
+            "[{},{},{}]",
+            self.network_id.to_string(),
```

Comment:
"{}" means to use the `Display` impl to format a type (vs. "{:?}" for the `Debug` impl). Since `NetworkId` and `RoleType` already impl `Display`, we can just use the `Display` impls for these fields

```rust
write!(f, "[{},{},{}]", self.network_id, self.role, self.peer_id.short_str())
```